### ISSUE TITLE
soho-application-menu not working with *ngIf

### ISSUE THREAD

**Issue by gustavpollak** [infor-design/enterprise-ng#1529]
**Describe the bug**
soho-application-menu is not working with *ngIf

**To Reproduce**

1. go onto enterprise-ng project and run it.
2. Edit code of a sample component to include following code sample:

```
<nav *ngIf="true" soho-application-menu >
  <div>
    <p>Test</p>
  </div>
</nav>
<button soho-nav-button></button>
```
3. Verify that clicking the nav-button does not trigger the toggle for the navigation menu.

**Expected behavior**
Navigation menu gets triggered by soho-nav-button click.

**Version**
- ids-enterprise-ng: 16.5.1

**Screenshots**
![ApplicationMenuToggleBrokengif](https://github.com/infor-design/enterprise-ng/assets/49652924/30ef01db-9566-4a45-8e5e-027080f9faa6)

**Platform**
 - Device: PC
 - OS Version: W11
 - Browser Name: Chrome 
 - Browser Version: Version 114.0.5735.199 (Official Build) (64-bit)

**Additional context**
Attached gif is in PortalV2 but I've tried the same in enterprise-ng project and can confirm it's the same issue.

**Comment by tmcconechy** [comment]
Yes, i see the problem. The current code in the click of the hamburger doesnt know the difference if its hidden with ngIf or not.

The only workaround i could think of would be to also disable or hide the hamburger button? Or disable the click code and add it back later? The problem is this code is internal.

Sort of an enhancement / sort of a bug / new use case
